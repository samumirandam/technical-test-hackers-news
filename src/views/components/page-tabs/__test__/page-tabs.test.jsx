/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/no-unresolved */
import React from 'react';
import { render, screen, fireEvent } from '@utils/test-utils';

import PageTabs from '../index';

const defaultProps = {};

const setup = (properties = {}) => {
  const setupStore = { ...defaultProps, ...properties };
  return render(<PageTabs {...setupStore} />, {});
};

describe('Test for PageTabs component', () => {
  test('Should render without errors', () => {
    setup();
    expect(screen.getByText('All')).toBeTruthy();
  });

  test('Should renders the same component', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  test('Should render click in All', () => {
    setup();
    expect(screen.getByText('All')).toBeTruthy();
    fireEvent.click(screen.getByText('All'));
    expect(global.window.location.pathname).toEqual('/');
  });

  test('Should render click in My faves', () => {
    setup();
    expect(screen.getByText('My faves')).toBeTruthy();
    fireEvent.click(screen.getByText('My faves'));
    expect(global.window.location.pathname).toEqual('/favorites');
  });
});
