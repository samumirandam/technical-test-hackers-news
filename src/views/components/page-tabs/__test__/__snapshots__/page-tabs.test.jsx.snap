// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for PageTabs component Should renders the same component 1`] = `
<div>
  <div
    class="PageTabs"
  >
    <button
      class="Button PageTabs__tab active"
      type="button"
    >
      All
    </button>
    <button
      class="Button PageTabs__tab"
      type="button"
    >
      My faves
    </button>
  </div>
</div>
`;
